src/dag.rs

```diff
@@ -30,43 +174,414 @@ impl<Types: RepoTypes> IpldDag<Types> {
         Ok(cid)
     }
 
-    pub async fn get(&self, path: IpfsPath) -> Result<Ipld, Error> {
+    /// Resolves a Cid rooted path to a document "node."
+    ///
+    /// Returns the resolved node as Ipld.
+    pub async fn get(&self, path: IpfsPath) -> Result<Ipld, ResolveError> {
+        // FIXME: do ipns resolve first
         let cid = match path.root().cid() {
             Some(cid) => cid,
-            None => return Err(anyhow::anyhow!("expected cid")),
+            None => panic!("not implemented: Ipns resolution, expected Cid based path"),
```

Comment:
mentioned in some commit that this was just changing panic message to another but I missed the earlier was using anyhow. well, *anyhow*.. good to fix when making this support ipns.